### DISCUSSION TITLE
Using Tasmota with script support and Shutters: How to update the variable sht[1] during Schutter is running?

### DISCUSSION THREAD

**Discussion by Sfre12** [arendst/Tasmota#19445] [Q&A] [answered]
Hey all.
I am using Tasmota to run my shutters. My Build is with script support.
Now I am displaying the Position of the shutters at the web interface by reading the variable sht[1] and sht[2]. This is working in general, but not during shutter is running. During running of the shutters, I can see the change of position in the console, but the variable sht[1] / [2] will not be updated.
Any idea how I can get that working/updating?

**Comment by stefanbode** [comment]
hi @Sfre12 : You should invest some time to get familiar with MQTT. In this case it is much easier to grab the values and use it. for example HomeAssistant is using TASMOTA and MQTT to show nice moving shutters in their UI

**Reply by Sfre12** [reply]
Thanks Stefan
I am Using MQTT with HomeAssistant and it is working, but my question was to have an update in UI of Tasmota device.
I think the answer from GEMU2015 seems to be the one I need .
Best regards Stephan F

**Comment by stefanbode** [comment]
On the HTML page the position is here:
```
const char HTTP_MSG_SLIDER_SHUTTER[] PROGMEM =
  "<div><span class='p'>%s</span><span class='q'>%s</span></div>"
  "<div><input type='range' min='0' max='100' value='%d' onchange='lc(\"u\",%d,value)'></div>";
```
the fist two % are "Open"/"Close". The %d at the value is what you're searching for. lc(u1..16) is the number of the shutter

**Reply by gemu2015** [reply, marked as answer]
you may read the position via MQTT in section >E
```
>D
pos=0
>E
pos=Shutter1#Position
print %pos%
```

**Reply by Sfre12** [reply]
Hey Gemu2015
Thank you very much. That what I was searching for!!

Please be so kind and check my other question ;-)
https://github.com/arendst/Tasmota/discussions/18647
